src/backend/api/meals.js

```diff
@@ -1,15 +1,199 @@
+          )
+        )
+        .groupBy('meal.id')
+        .having(
+          knex.raw(
+            '(max_reservations > COALESCE(SUM(reservation.number_of_guests),0))'
+          )
+        )
+        .groupBy('meal.id')
+    }
+    response.json(meals)
+  } catch (error) {
+    throw error
+  }
+})
+
+//GET. Returns meal by id
+router.get('/:id', async (request, response) => {
+  try {
+    const mealId = Number(request.params.id)
+    if (isNaN(mealId)) {
+      response.status(400).json({
+        error: "Meal wasn't found.",
+        hint: 'Id should be a number',
+      })
+      return
+    } else {
+      const meals = await knex('meal')
+      const isFoundId = meals.some((meal) => meal.id === mealId)
+      if (!isFoundId) {
+        response.status(400).json({
+          error: "Meal wasn't found.",
+          hint: 'Meal with this id does not exist',
+        })
+        return
+      }
+      const mealById = await knex('meal').select('*').where({ id: mealId })
+      response.json(mealById)
+    }
+  } catch (error) {
+    throw error
+  }
+})
+
+//POST.	Adds a new meal
+router.post('/', async (request, response) => {
+  try {
+    const context = request.body
+    if (Object.keys(context).length === 0 || context === {}) {
+      response.statusCode(422).json({ message: 'Bad input' })
+      return
+    }
+    if (context.title === '') {
+      response.json({ message: 'Give me title' })
+      return
+    }
+    if (context.description === '') {
+      response.json({ message: 'Give me description' })
+      return
+    }
+    if (context.location === '') {
+      response.json({ message: 'Give me location' })
+      return
+    }
+    if (context.when === '') {
+      response.json({ message: 'Give me date' })
+      return
+    }
+    if (context.max_reservations === '') {
+      response.json({ message: 'Give number of guests' })
+      return
+    }
+    if (context.price === '') {
+      response.json({ message: 'Give me max price' })
+      return
+    }
+    if (context.created_date === '') {
+      response.json({ message: 'Give me created date' })
+      return
+    }
+    const insertContext = await knex('meal').insert(context)
+    response.json(insertContext)
+  } catch (error) {
+    throw error
+  }
+})
+
+// PUT.	Updates the meal by id
+router.put('/:id', async (request, response) => {
+  try {
+    const mealId = Number(request.params.id)
+    if (isNaN(mealId)) {
+      response.status(400).json({
+        error: "Meal wasn't changed.",
+        hint: 'Id should be a number',
+      })
+      return
+    } else {
+      const meals = await knex('meal')
+      const isFoundId = meals.some((meal) => meal.id === mealId)
+      if (!isFoundId) {
+        response.status(400).json({
+          error: "Meal wasn't changed.",
+          hint: 'Meal with this id does not exist',
+        })
+        return
+      }
```

Comment:
Same comment about this as previously. I don't think this is needed, but maybe you can explain what was your reasoning about it

Reply:
The reason was on a first step just to check if searching item exists. To prevent undesirable interaction with database. On a second step to make a certain manipulations on database data